source/dub/generators/visuald.d

```diff
@@ -141,7 +141,8 @@ class VisualDGenerator : ProjectGenerator {
 		bool isHeaderOnlyPackage(string pack, in TargetInfo[string] targets)
 		const {
 			auto buildsettings = targets[pack].buildSettings;
-			if (!buildsettings.sourceFiles.any!(f => f.endsWith(".d"))())
+			if (!buildsettings.sourceFiles.any!(f => f.endsWith(".d"))() &&
+				!buildsettings.sourceFiles.any!(f => f.endsWith(".c"))() )
```

Comment:
Replace with

```d
if (!buildsettings.sourceFiles.any!(f => f.endsWith(".d", ".c")))
```

.

Reply:
can we move the list of hardcoded file endings somewhere else entirely? It's repeated many times in the code.